internal/jobservice/repository/sql_job_service.go

```diff
@@ -125,17 +125,19 @@ func jobStateStrToJSRState(jobState string) (js.JobServiceResponse_State, error)
 
 // Update database with JobTable.
 func (s *SQLJobService) UpdateJobServiceDb(jobTable *JobStatus) {
+	// SQLite only allows one write at a time. Therefore we must serialize
+	// writes in order to avoid SQL_BUSY errors.
+	s.writeLock.Lock()
+	defer s.writeLock.Unlock()
+
 	stmt, err := s.db.Prepare("INSERT OR REPLACE INTO jobservice VALUES (?, ?, ?, ?, ?, ?)")
 	if err != nil {
 		panic(err)
 	}
+	defer stmt.Close()
```

Comment:
I also added this since, according to go docs, you should close prepared statements once you're done with them. We may want to look into preparing these statements once during DB creation and then re-using them.